syne_tune/search_space.py

```diff
@@ -456,6 +489,11 @@ def cast(self, value):
     def match_string(self, value) -> str:
         return str(self.categories.index(value))
 
+    def __eq__(self, other) -> bool:
+        return isinstance(other, Categorical) \
+               and super(Categorical, self).__eq__(other) \
+               and self.categories == other.categories
```

Comment:
Do we care about the order of categories? We might want to use set equality instead.